controllers/managedosimage_controller.go

```diff
@@ -130,14 +131,22 @@ func (r *ManagedOSImageReconciler) reconcile(ctx context.Context, managedOSImage
 		return ctrl.Result{}, fmt.Errorf("failed to create fleet bundle resources: %w", err)
 	}
 
-	if err := r.createFleetBundle(ctx, managedOSImage, bundleResources); err != nil {
-		meta.SetStatusCondition(&managedOSImage.Status.Conditions, metav1.Condition{
-			Type:    elementalv1.FleetBundleCreation,
-			Reason:  elementalv1.FleetBundleCreateFailureReason,
-			Status:  metav1.ConditionFalse,
-			Message: err.Error(),
-		})
-		return ctrl.Result{}, fmt.Errorf("failed to create fleet bundle: %w", err)
+	// Create a new Fleet bundle if we didn't do it before. Otherwise update it.
+	if meta.IsStatusConditionTrue(managedOSImage.Status.Conditions, elementalv1.FleetBundleCreation) {
```

Comment:
Given that there was this condition already in place I wouldn't be surprised this was the expected behavior from the very beginning and probably it was even coded this way (at the time the operator was coded as a wrangler controller) . Probably this was a regression introduced at the time we switched to runtime-controller framework.